scripts/src/java/org/oppia/android/scripts/label/AccessibilityLabelCheck.kt

```diff
@@ -0,0 +1,156 @@
+
+  val accessibilityLabelExemptiontextProto = "scripts/assets/accessibility_label_exemptions"
+
+  val accessibilityLabelExemptionList = loadAccessibilityLabelExemptionsProto(
+    accessibilityLabelExemptiontextProto
+  ).getExemptedActivityList()
+
+  val manifesFilePath = args[1]
+
+  val fullPathToManifestFile = repoPath + manifesFilePath
+
+  val builderFactory = DocumentBuilderFactory.newInstance()
+
+  val docBuilder = builderFactory.newDocumentBuilder()
+
+  val doc = docBuilder.parse(File(fullPathToManifestFile))
+  doc.getDocumentElement().normalize()
+
+  val completeActivityList = convertNodeListToListOfNode(doc.getElementsByTagName("activity"))
+
+  val activityListWithoutLabel = completeActivityList.filter { node ->
+    checkIfActivityHasMissingLabel(node, accessibilityLabelExemptionList)
+  }
+
+  logFailures(activityListWithoutLabel, repoPath, accessibilityLabelExemptiontextProto)
+
+  if (activityListWithoutLabel.isNotEmpty()) {
+    throw Exception("ACCESSIBILITY LABEL CHECK FAILED")
+  } else {
+    println("ACCESSIBILITY LABEL CHECK PASSED")
+  }
+}
+
+/**
+ * Checks whether an activity element has a missing label.
+ *
+ * @param activityNode instance of Node
+ * @param accessibilityLabelExemptionList list of all the exemptions of the label check
+ * @return whether the label is present or not
+ */
+private fun checkIfActivityHasMissingLabel(
+  activityNode: Node,
+  accessibilityLabelExemptionList: List<String>
+): Boolean {
+  val attributesList = activityNode.getAttributes()
+  val activityPath = attributesList.getNamedItem("android:name").getNodeValue()
+  if (
+    activityPath
+      .removePrefix(".")
+      .replace(".", "/") in accessibilityLabelExemptionList
+  ) {
+    return false
+  }
+  return attributesList.getNamedItem("android:label") == null
+}
+
+/**
+ * The [nodeList] is not iterable. This helper
+ * function converts it to List<Node>.
+ *
+ * @param nodeList instance of NodeList
+ * @return a list of nodes
+ */
+private fun convertNodeListToListOfNode(nodeList: NodeList): List<Node> {
+  return IntStream.range(0, nodeList.getLength())
+    .mapToObj(nodeList::item)
+    .collect(Collectors.toList())
+}
+
+/**
+ * Logs the failures for accessibility label check.
+ *
+ * @param matchedNodes a list of nodes having missing label
+ * @param repoPath path of the repo to be analyzed
+ */
+private fun logFailures(
+  matchedNodes: List<Node>,
+  repoPath: String,
+  accessibilityLabelExemptiontextProto: String
+) {
+  val pathPrefix = "${repoPath}app/src/main/java/org/oppia/android"
+  if (matchedNodes.isNotEmpty()) {
+    println("Accessiblity labels missing for Activities:")
+    matchedNodes.sortedBy {
+      it.getAttributes()
+        .getNamedItem("android:name")
+        .getNodeValue()
+    }.forEach { node ->
+      println(
+        "- $pathPrefix" +
+          "${
+          node.getAttributes()
+            .getNamedItem("android:name")
+            .getNodeValue()
+            .replace(".", "/")
```

Comment:
Please de-duplicate with the code above doing the same thing (I suggest using a data class or helper to avoid needing duplicate the code in both places).

Reply:
This is outdated.